lib/puppet-languageserver/puppetfile/validation_provider.rb

```diff
@@ -45,6 +45,34 @@ def self.validate(content, _max_problems = 100)
                                         'message'  => mod.properties[:error_message])
         end
 
+        # Check for invalid module definitions
+        require 'puppet_forge'
+        puppetfile.modules.each do |puppet_module|
+          if puppet_module.class == R10K::Module::Forge
+
+            module_name       = puppet_module.title.gsub('/', '-')
+            installed_version = puppet_module.version
+
+            begin
+              forge_module      = PuppetForge::Module.find(module_name)
+            rescue StandardError, SyntaxError, LoadError => e
```

Comment:
Besides using the correct catch here, this is intended to handle no internet. Should have an option to disable this check